osu.Game/OsuGame.cs

```diff
@@ -898,8 +899,41 @@ private void handleStartupImport()
             if (args?.Length > 0)
             {
                 string[] paths = args.Where(a => !a.StartsWith('-')).ToArray();
+
                 if (paths.Length > 0)
-                    Task.Run(() => Import(paths));
+                {
+                    string firstPath = paths.First();
+
+                    if (firstPath.StartsWith(OSU_PROTOCOL, StringComparison.Ordinal))
+                    {
+                        handleOsuProtocolUrl(firstPath);
+                    }
+                    else
+                    {
+                        Task.Run(() => Import(paths));
+                    }
+                }
+            }
+        }
+
+        private void handleOsuProtocolUrl(string url)
+        {
+            if (!url.StartsWith(OSU_PROTOCOL, StringComparison.Ordinal))
+                throw new ArgumentException("Invalid osu URL provided.", nameof(url));
+
+            string[] pieces = url.Split('/');
```

Comment:
I initially wanted to propose using `StringSplitOptions.RemoveEmptyEntries` here, but I guess not specifying it makes the indexes below also implicitly check that the URL is formed correctly (i.e. with removing empty entries, `osu:///////s/1` would also be accepted). Maybe best to pre-empt that with an inline comment?

Reply:
I'm not really sure how to phrase the inline comment, but am open to adding one if you word it.

Reply:
I've pushed an inline comment in - please check whether you find it legible.